docker/with_the_same_user

```diff
@@ -36,7 +36,12 @@ else
   rm /this_is_writable_file_system
 fi
 
-getent group "${CI_BUILD_GID}" || addgroup --force-badname --gid "${CI_BUILD_GID}" "${CI_BUILD_GROUP}"
+getent group "${CI_BUILD_GID}" || (
+    # Ensure "${CI_BUILD_GROUP}" is not already some other gid inside container.
+    if grep -q "^${CI_BUILD_GROUP}:" /etc/group; then
+        CI_BUILD_GROUP="${CI_BUILD_GROUP}2"
```

Comment:
rather than do this, should we just set CI_BUILD_GROUP to the existing group? i think having two groups with identical gid is a bit weird, though cursory google does say it's supported. just curious if you tried this and didn't like that appraoch?

Reply:
The `CI_BUILD_GROUP` already exists, but has a different GID. I thought it was important to keep the same GID and that's why we create a new CI_BUILD_GROUP name and use that with the same GID, but there is just one group with the given `CI_BUILD_GID`.

We can alternatively change the `CI_BUILD_GID` to the one that's mapped to `CI_BUILD_GROUP`, but I thought that doesn't work. Please let me know if my understanding is mistaken here.